test/inductor/test_perf.py

```diff
@@ -1241,7 +1241,7 @@ def scaled_index_add(x, y, scale_y):
             return out
 
         inp = (T(10, 10), T(5, 10), T(10))
-        self.assertExpectedInline(count_numel(scaled_index_add, *inp), """250""")
+        self.assertExpectedInline(count_numel(scaled_index_add, *inp), """370""")
```

Comment:
I am afraid we have to take these extra buffer reads/writes since although it works for this case due to there is no index overlapping, if we slightly change the case as in `test_cpu_repo.py::test_index_add`, the result is still in-correct.

Reply:
Hi @Chillee, do you think it's ok adding extra reads/writes for this UT? Seems you added this UT with a special `randperm_index_add_pattern` replacement: https://github.com/pytorch/pytorch/blob/6ef6b3f586c6f7b06c6b3b1bf7c264cc7f044687/torch/_inductor/fx_passes/misc_patterns.py#L33-L52

And I feels difficult to figure out the index comes from `randperm` and completely unique at the lowering phase with this PR where we fix another issue.

Reply:
I don't think we should pessimize the compiler to exactly match eager for undefined behavior. If it's undefined behavior, I think we should just "won't fix".

Reply:
Even though we saw this issue for repeated indices which is undefined behaviour, I don't think it is unique to duplicated indices. Are we sure that this issue happens only for repeated indices?

Reply:
yean, just add a new UT: `test_index_put2` in which the index is: `torch.tensor([[0, 1]])` and `torch.tensor([[1, 0]])` which has no repeated indices, but the test case will also fail wo the fix. Due to that self and values have `weakdep`.

Reply:
Does that new test case fail for inductor on gpu ?

Reply:
This example code is also only an issue with reinplace_inplaceable_ops cc @Chillee